tests/test_api_endpoints.py

```diff
@@ -440,3 +411,73 @@ def test_openapi_endpoint():
             headers={"x-rh-identity": auth_token}
         )
         assert response.json == content_from_file
+
+
+def get_rbac_mock_file(filename):
+    with open(f"{Path(__file__).parent}/data_files/{filename}") as f:
+        json_data = json.loads(f.read())
+    return json_data
+
+
+def mock_enable_rbac(mocker):
+    mocker.patch('ros.lib.rbac_interface.ENABLE_RBAC', return_value=True)
+
+
+def mock_rbac(json_data, mocker):
+    mocker.patch('ros.lib.rbac_interface.query_rbac', return_value=json_data)
+
+
+def test_systems_rbac_returns_groups_including_example_group(
+        auth_token,
+        db_setup,
+        db_create_account,
+        db_create_system,
+        system_with_example_group,
+        system_with_test_group,
+        db_create_performance_profile,
+        create_performance_profiles,
+        mocker):
+    with app.test_client() as client:
+        mock_enable_rbac(mocker)
+        mock_rbac(get_rbac_mock_file("mock_rbac_returns_groups_including_example_group.json"), mocker)
+        response = client.get('/api/ros/v1/systems', headers={"x-rh-identity": auth_token})
+        assert response.status_code == 200
+        assert response.json["meta"]["count"] == 2
+        assert response.json["data"][0]["groups"][0]["name"] == "example-group"
+
+
+def test_systems_rbac_returns_emtpy_group(
+        auth_token,
+        db_setup,
+        db_create_account,
+        db_create_system,
+        system_with_example_group,
+        system_with_test_group,
+        db_create_performance_profile,
+        create_performance_profiles,
+        mocker):
+    with app.test_client() as client:
+        mock_enable_rbac(mocker)
+        mock_rbac(get_rbac_mock_file("mock_rbac_returns_emtpy_group.json"), mocker)
+        response = client.get('/api/ros/v1/systems', headers={"x-rh-identity": auth_token})
+        assert response.status_code == 200
+        assert response.json["meta"]["count"] == 1
+        assert response.json["data"][0]["groups"] == []
+
+
+def test_systems_mock_rbac_returns_no_groups(
+        auth_token,
+        db_setup,
+        db_create_account,
+        db_create_system,
+        system_with_example_group,
+        system_with_test_group,
+        db_create_performance_profile,
+        create_performance_profiles,
+        mocker):
+    with app.test_client() as client:
+        mock_enable_rbac(mocker)
+        mock_rbac(get_rbac_mock_file("mock_rbac_returns_no_groups.json"), mocker)
+        response = client.get('/api/ros/v1/systems', headers={"x-rh-identity": auth_token})
+        assert response.status_code == 200
+        assert response.json["meta"]["count"] == 3
```

Comment:
@upadhyeammit do you think we should add comments above each test case to understand what scenario it is testing? like what will be expected output to a particular scenario?